### PR TITLE
[core] Add "--debug" flag for CPD

### LINKED ISSUE #3796: [core] CPD should also provide a `--debug` flag
**Is your feature request related to a problem? Please describe.**
When running PMD from command line, you can enable debug logging with `--debug`. This feature is missing for CPD.
This is inconsistent.

**Describe the solution you'd like**
Add `--debug` for CPD with the same semantics like PMD.

**Describe alternatives you've considered**
none

**Additional context**
- Found in #3781 where we suggest to run with "--debug" to see any stack traces.

### PR DESCRIPTION
**Pull request by adangel**
## Describe the PR

- Also adds the alternative `--verbose`, but doesn't add the single character options.
- Adds a couple of debug log statements to CPD - currently we do not seem to use logging very often here - usually a System.err print is used. But that's something for later to improve.
- Fixes #3796

## Ready?

- [x] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [x] Added (in-code) documentation (if needed)

### CODE REVIEW

File: pmd-core/src/test/java/net/sourceforge/pmd/cpd/CPDCommandLineInterfaceTest.java [hunk 1]
```diff
@@ -80,12 +85,29 @@ public void testDeprecatedOptionsWarning() throws IOException {
...
         assertFalse(loggingRule.getLog().contains("Some deprecated options were used on the command-line, including --filelist"));
         assertFalse(loggingRule.getLog().contains("Consider replacing it with --file-list"));
     }
+
+    @Test
+    public void testDebugLogging() {
+        CPD.StatusCode statusCode = CPD.runCpd("--minimum-tokens", "340", "--language", "java", "--files",
+                SRC_DIR, "--debug");
+        assertEquals(CPD.StatusCode.OK, statusCode);
+        assertTrue(errLog.getLog().contains("Tokenizing ")); // this is a debug logging
```
Comment (oowekyala) [inline pmd-core/src/test/java/net/sourceforge/pmd/cpd/CPDCommandLineInterfaceTest.java:102]: Please use `MatcherAssert.assertThat` and not assertTrue (other tests here don't, but assertTrue gives terrible error messages...)
```suggestion
        assertThat(errLog.getLog(), containsString("Tokenizing ")); // this is a debug logging
```

File: pmd-core/src/test/java/net/sourceforge/pmd/cpd/CPDCommandLineInterfaceTest.java [hunk 2]
```diff
@@ -80,12 +85,29 @@ public void testDeprecatedOptionsWarning() throws IOException {
...
+    }
+
+    @Test
+    public void testNormalLogging() {
+        loggingRule.clear();
+        CPD.StatusCode statusCode = CPD.runCpd("--minimum-tokens", "340", "--language", "java", "--files",
+                SRC_DIR);
+        assertEquals(CPD.StatusCode.OK, statusCode);
+        assertFalse(errLog.getLog().contains("Tokenizing ")); // this is a debug logging
```
Comment (oowekyala) [inline pmd-core/src/test/java/net/sourceforge/pmd/cpd/CPDCommandLineInterfaceTest.java:111]:
```suggestion
        assertThat(errLog.getLog(), not(containsString("Tokenizing "))); // this is a debug logging
```

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 0 violations,<br />
introduces 0 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/160a89b1362944fdb82a8046dc9ddcbb/diff1/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by oowekyala — APPROVED** [review]
Thanks!

**Pull request merged**